### PR TITLE
Declaration evaluation

### PR DESCRIPTION
**Pull request by Horneth**
Coerce evaluated declaration to their expected type + Bonus wdl4s update to fix https://github.com/broadinstitute/centaur/pull/176

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ before_install:
   - openssl aes-256-cbc -K "$encrypted_5ebd3ff04788_key" -iv "$encrypted_5ebd3ff04788_iv" -in src/bin/travis/resources/jesConf.tar.enc -out jesConf.tar -d || true
 env:
   global:
-    - CENTAUR_BRANCH=develop
+    - CENTAUR_BRANCH=mapComplexType
```
Comment (Horneth) [inline .travis.yml]: revert before merging

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala [hunk 2]
```diff
@@ -764,7 +766,11 @@ object WorkflowExecutionActor {
   
   case class StaticDeclarationKey(scope: Declaration, index: ExecutionIndex, value: WdlValue) extends DeclarationKey
   
-  case class DynamicDeclarationKey(scope: Declaration, index: ExecutionIndex, requiredExpression: WdlExpression) extends DeclarationKey
+  case class DynamicDeclarationKey(scope: Declaration, index: ExecutionIndex, requiredExpression: WdlExpression) extends DeclarationKey {
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala]: ToL: `scope` seems an odd variable name for a declaration!

**Comment by cjllanwarne** [comment]
LGTM 👍

**Comment by mcovarr** [comment]
:+1:

**Pull request merged**